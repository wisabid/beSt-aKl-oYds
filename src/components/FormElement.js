import React from 'react';

export const InputEl = (props) => {
    return (
        <fieldset>
            <legend>{props.legend}</legend>
            <input type={props.eltype} placeholder={props.elplaceholder} id={props.elname} autoComplete="off" />
        </fieldset>
    )
}

export const ButtonEl = (props) => {
    return (
        <button type={props.eltype} onClick={props.click}>{props.elvalue}</button>
    )
}

