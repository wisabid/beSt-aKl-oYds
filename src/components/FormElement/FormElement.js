import React from 'react';
import PropTypes from 'prop-types';

export const InputEl = (props) => {
    return (
        <fieldset>
            <legend>{props.legend}</legend>
            <input type={props.eltype} placeholder={props.elplaceholder} id={props.elname} autoComplete="off" />
        </fieldset>
    )
}

InputEl.propTypes = {
    legend: PropTypes.string.isRequired,
    eltype: PropTypes.string.isRequired, 
    elplaceholder: PropTypes.string,
    elname: PropTypes.string.isRequired
}

export const ButtonEl = (props) => {
    return (
        <button type={props.eltype} onClick={props.click}>{props.elvalue}</button>
    )
}

ButtonEl.propTypes = {
    eltype: PropTypes.string.isRequired,
    elvalue:PropTypes.string.isRequired
}

