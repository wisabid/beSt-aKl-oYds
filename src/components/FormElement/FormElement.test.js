import React from 'react';
import { shallow } from 'enzyme';
import { InputEl, ButtonEl } from './FormElement';

describe('<FormElement />', () => {
    const InputElComp = shallow(<InputEl legend="Name" elname="username" eltype="text" elplaceholder="Enter your Name"/>);
    const ButtonComp = shallow(<ButtonEl elvalue="Go" />);
    it('<InputEl /> should match the snapshot', () => {
        expect(InputElComp).toMatchSnapshot();
    })

    it('<ButtonEl /> should match the snapshot', () => {
        expect(ButtonComp).toMatchSnapshot();
    })
})