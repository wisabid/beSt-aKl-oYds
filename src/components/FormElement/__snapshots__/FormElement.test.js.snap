// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormElement /> <ButtonEl /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonEl
    elvalue="Go"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Go",
      "onClick": undefined,
      "type": undefined,
    },
    "ref": null,
    "rendered": "Go",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Go",
        "onClick": undefined,
        "type": undefined,
      },
      "ref": null,
      "rendered": "Go",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<FormElement /> <InputEl /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputEl
    elname="username"
    elplaceholder="Enter your Name"
    eltype="text"
    legend="Name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <legend>
          Name
        </legend>,
        <input
          autoComplete="off"
          id="username"
          placeholder="Enter your Name"
          type="text"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
        },
        "ref": null,
        "rendered": "Name",
        "type": "legend",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "id": "username",
          "placeholder": "Enter your Name",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "fieldset",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <legend>
            Name
          </legend>,
          <input
            autoComplete="off"
            id="username"
            placeholder="Enter your Name"
            type="text"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
          },
          "ref": null,
          "rendered": "Name",
          "type": "legend",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "id": "username",
            "placeholder": "Enter your Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "fieldset",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
